src/main/java/org/jabref/logic/importer/fetcher/SemanticScholar.java

```diff
@@ -0,0 +1,267 @@
+            LOGGER.debug(response.toString());
+            if (response.isEmpty()) {
+                return Collections.emptyList();
+            }
+
+            int total = response.getInt("total");
+            if (total == 0) {
+                return Collections.emptyList();
+            } else if (response.has("next")) {
+                total = Math.min(total, response.getInt("next") - response.getInt("offset"));
+            }
+
+            // Response contains a list
+            JSONArray items = response.getJSONArray("data");
+            List<BibEntry> entries = new ArrayList<>(items.length());
+            for (int i = 0; i < total; i++) {
+                JSONObject item = items.getJSONObject(i);
+                BibEntry entry = jsonItemToBibEntry(item);
+                entries.add(entry);
+            }
+
+            return entries;
+        };
+    }
+
+    /**
+     * This is copy-paste from CrossRef, need to be checked.
+     *
+     * @param item an entry received, needs to be parsed into a BibEntry
+     * @return The BibEntry that corresponds to the received object
+     */
+    private BibEntry jsonItemToBibEntry(JSONObject item) throws ParseException {
+        try {
+            BibEntry entry = new BibEntry(StandardEntryType.Article);
+            entry.setField(StandardField.URL, item.optString("url"));
+            entry.setField(StandardField.TITLE, item.optString("title"));
+            entry.setField(StandardField.ABSTRACT, item.optString("abstract"));
+            entry.setField(StandardField.VENUE, item.optString("venue"));
+            entry.setField(StandardField.YEAR, item.optString("year"));
+
+            entry.setField(StandardField.AUTHOR, toAuthors(item.optJSONArray("authors")));
+
+            // TODO: check. I'm not sure about the EPRINT and arXiv fields
+            JSONObject externalIds = item.optJSONObject("externalIds");
+            entry.setField(StandardField.DOI, externalIds.optString("DOI"));
+            if (externalIds.has("ArXiv")) {
+                entry.setField(EPRINT, externalIds.getString("ArXiv"));
+                entry.setField(StandardField.ARCHIVEPREFIX, "arXiv");
+            }
+            entry.setField(StandardField.PMID, externalIds.optString("PubMed"));
+//            exIds.addAll(externalIds.keySet());
+            return entry;
+        } catch (JSONException exception) {
+            throw new ParseException("SemanticScholar API JSON format has changed", exception);
+        }
+    }
+
+    private String toAuthors(JSONArray author) {
+        if (author == null || author.isEmpty()) {
+            return "";
+        }
+        StringBuilder authors = new StringBuilder();
+        for (int i = 0; i < author.length(); i++) {
+            JSONObject item = author.getJSONObject(i);
+            authors.append(item.getString("name")).append(" and ");
+        }
+        authors.delete(authors.length() - 5, authors.length());
```

Comment:
I would use a StringJoiner https://www.baeldung.com/java-string-joiner  together with a stream. Yoj can take a look at MedraFetcher getAuthors and use it like this
```
   return IntStream.range(0, authors.length())
                        .mapToObj(authors::getJSONObject)
  .map((author) -> author.has("name") ?  author.getString("name")
  .collect(Collectors.joininng("and)) 
```
Not sure if necessary, but additionally you could also call then Authorlist.parse() on that result. It parses all kinds of strings into authors and then return it in a correct order. But this might also be handled in a post cleanup.